UnitySetup/UnitySetup.psm1

```diff
@@ -2132,3 +2130,537 @@ function Get-UnityLicense {
         Write-Warning "Alias $($_.Name) already configured by $($alias.Source)"
     }
 }
+
+function Import-UPMConfig {
+    [CmdletBinding()]
+    param(
+        [Parameter()]
+        [String]$ProjectManifestPath,
+        [int]$SearchDepth = 3
+    )
+
+    $ProjectManifestPaths = @()
+
+    if ((Get-Item $ProjectManifestPath) -is [System.IO.DirectoryInfo]) {
+        Write-Host "Path provided is not a manifest.json file ($ProjectManifestPath), will attempt search"
+        $FoundPaths = @(Get-Childitem -Path $ProjectManifestPath -Include manifest.json -File -Recurse -Depth $SearchDepth -ErrorAction SilentlyContinue)
+        foreach ($file in $FoundPaths) {
+            $ProjectManifestPaths += $file.FullName
+            if ($Verbose) { Write-Host "Found ($file.FullName)" }
+        }
+    }
+    else {
+        $ProjectManifestPaths += $ProjectManifestPath
+    }
+
+    if (([string]::IsNullOrEmpty($ProjectManifestPath)) -or (-not $(Test-Path $ProjectManifestPath))) {
+        Write-Error "Unable to find manifest.json file, please provide a path pointing directly to a Unity project's manifest.json or provide a path to a Unity project"
+        return @()
+    }
+
+    $scopedRegistrySet = [System.Collections.Generic.HashSet[string]]::new()
+    foreach ($ManifestPath in $ProjectManifestPaths) {
+        $manifest = Get-Content -Path $ManifestPath | ConvertFrom-Json
+
+        foreach ($scopedRegistry in $manifest.scopedRegistries) {
+            $url = $scopedRegistry.url -replace '/$', ''  
+            if ($url -like 'https://pkgs.dev.azure.com/*') {
```

Comment:
Definitely think we want to include all scoped registries in the object and not just ADO ones. Perhaps we add a dynamic property to the scoped registries to compute/return if it's an ADO one or not?

Reply:
So... about this.  If we accept all scoped registries, do we know if we'll we be able to get the auth tokens / confirm the token for those registries?

When retrieving and validating PATs we rely on hard coded (albeit wildcard replaced) URLs for dev.azure.com and feeds.dev.azure.com to fetch and validate tokens.  I will do a bit of research into this, but I'm not sure if a scoped registry URL provides enough information by itself for us to fetch / verify a PAT.  My expertise in this area is pretty slim, though.

Reply:
I mean more that the import cmdlet should include them all - it would make sense to me when using those scoped registries to 'Sync' that we'd handle the ADO ones and we'd maybe warn or log that for ones that we didn't recognize.